llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -639,7 +639,7 @@ bool AMDGPUInstructionSelector::selectG_MERGE_VALUES(MachineInstr &MI) const {
   LLT SrcTy = MRI->getType(MI.getOperand(1).getReg());
 
   const unsigned SrcSize = SrcTy.getSizeInBits();
-  if (SrcSize < 32)
+  if (SrcSize < 16)
```

Comment:
I'm not sure I understand why this is selectImpl. This is just a fancy way of failing? Is this incorrectly passing selection for non-true16 targets?

Reply:
I double checked this. It seems this is never invoked at all when srcSize is smaller than 32.

we might just don't need this patch